maint/local_python/binding_c.py

```diff
@@ -891,15 +892,9 @@ def dump_abi_wrappers(func, is_large):
     is_neighbor_alltoallw = False
     if re.match(r'mpi_i?alltoallw(_init)?', func['name'], re.IGNORECASE):
         array_type = "Datatype"
-        array_size = "comm_size"
+        array_size = "peer_size"
         is_alltoallw = True
-        pre_filters.append("MPI_Comm comm = ABI_Comm_to_mpi(comm_abi);")
-        pre_filters.append("int comm_size = 0;")
-        pre_filters.append("if (comm != MPI_COMM_NULL) {")
-        pre_filters.append("    MPIR_Comm *comm_ptr;")
-        pre_filters.append("    MPIR_Comm_get_ptr(comm, comm_ptr);")
-        pre_filters.append("    comm_size = (comm_ptr->comm_kind == MPIR_COMM_KIND__INTERCOMM) ? comm_ptr->remote_size : comm_ptr->local_size;")
```

Comment:
Technically no, but if you look at the implementation of `MPI_Comm_size`, things are not so trivial as using `comm_ptr->local_size`. Therefore, I though that using `PMPI_` routines everywhere is the safest and future-proof way to do it.

Reply:
`PMPI_` routines are user-facing, they may have error checking or code logics that are essentially tricks such as ABI swaps. `Comm_size` is a trivial case that doesn't have such complications, but even worrying about it adds to the development/maintenance cost.  So as a principle, `PMPI_` routines are only called by `MPI_` layer. There may be rule-breakers in our current code base; we need to eliminate them.

If the code is not simple, create and use internal macros or functions. For example, most `PMPI_` routines call `MPIR_` functions.

Reply:
Threadcomm is an MPICH-specific extension and thus not supported via abi yet. It requires more logic in the binding layer to support threadcomm in communication routines than just fix the rank/size. You can use `MPIR_Comm_size_impl` instead of `PMPI_Comm_size`. But again, I would like to establish that it is broken first before fixing it. Establishing principles/rules also count as establishing broken, but we need cover cases throughout the code base and a clear statement for applying new rules. We internally direct access `comm->size/rank` everywhere, so I am not going to fix some of them unless we establish rules.

Reply:
OK, let me add a commit removing the use of PMPI APis while trying to be resilient to garbage input arguments.